src/ocm/components/clusterConfiguration/ReviewFeatureSupportLevel.tsx

```diff
@@ -0,0 +1,104 @@
+import { CheckCircleIcon, InfoCircleIcon } from '@patternfly/react-icons';
+import React from 'react';
+import { FeatureSupportLevelContext } from '../../../common/contexts';
+import { global_success_color_100 as okColor } from '@patternfly/react-tokens';
+import { TextList, TextListItem, TextContent } from '@patternfly/react-core';
+import {
+  FeatureId,
+  SupportLevelMap,
+  PreviewSupportLevel,
+  isPreviewSupportLevel,
+} from '../../../common/types';
+import { SUPPORT_LEVEL_LINKS } from '../../../common/config/constants';
+import ExternalLink from '../../../common/components/ui/ExternalLink';
+
+export type SupportLevelSummary = {
+  unsupportedVms: boolean;
+  featureIds: FeatureId[];
+  supportLevel: PreviewSupportLevel;
+};
+
+const getFeatureReviewText = (featureId: FeatureId): string => {
+  switch (featureId) {
+    case 'SNO':
+      return 'Install single node OpenShift (SNO)';
+    case 'VIP_AUTO_ALLOC':
+      return 'Allocate virtual IPs via DHCP server';
+    default:
+      return featureId;
+  }
+};
+
+const getPreviewSupportLevelTitle = (supportLevel: PreviewSupportLevel) => {
+  const linkText: string =
+    supportLevel === 'dev-preview' ? 'Developer Preview features' : 'Technology Preview Features';
+
+  return <ExternalLink href={SUPPORT_LEVEL_LINKS[supportLevel]}>{linkText}</ExternalLink>;
+};
+
+const getPreviewFeatureList = (supportLevelMap: SupportLevelMap) => {
+  const previewSupportLevels: { [key in PreviewSupportLevel]: FeatureId[] } = {
+    'tech-preview': [],
+    'dev-preview': [],
+  };
+  for (const [featureId, supportLevel] of Object.entries(supportLevelMap)) {
+    if (!isPreviewSupportLevel(supportLevel)) {
+      continue;
+    }
+    previewSupportLevels[supportLevel].push(featureId as FeatureId);
+  }
+  let supportLevel: PreviewSupportLevel;
+  //Show only one preview support level list, first priority to developer preview features
+  if (previewSupportLevels['dev-preview'].length) {
+    supportLevel = 'dev-preview';
+  } else if (previewSupportLevels['tech-preview'].length) {
+    supportLevel = 'tech-preview';
+  } else {
+    return null;
+  }
+  const featureList = previewSupportLevels[supportLevel].map((featureId: FeatureId) => (
```

Comment:
`supportLevel` can be `null` --> `previewSupportLevels[supportLevel]` can be `undefined` --> `previewSupportLevels[supportLevel].map(...)` throws: `Uncaught TypeError: Cannot read properties of undefined (reading 'map')`
```suggestion
  const featureList = previewSupportLevels[supportLevel].map((featureId: FeatureId) => (
```

Reply:
The support levels that are not 'tech-preview' and 'dev-preview' are filtered in line 45:
` for (const [featureId, supportLevel] of Object.entries(supportLevelMap)) {
           if (!isPreviewSupportLevel(supportLevel)) {
`